x-pack/plugins/cloud_security_posture/server/create_indices/create_indices.ts

```diff
@@ -7,30 +7,28 @@
 import type { ElasticsearchClient, Logger } from '@kbn/core/server';
 import { errors } from '@elastic/elasticsearch';
 import type { MappingTypeMapping } from '@elastic/elasticsearch/lib/api/types';
+
 import {
   BENCHMARK_SCORE_INDEX_DEFAULT_NS,
   BENCHMARK_SCORE_INDEX_PATTERN,
   BENCHMARK_SCORE_INDEX_TEMPLATE_NAME,
   CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
-  FINDINGS_INDEX_NAME,
-  LATEST_FINDINGS_INDEX_DEFAULT_NS,
-  LATEST_FINDINGS_INDEX_PATTERN,
-  LATEST_FINDINGS_INDEX_TEMPLATE_NAME,
 } from '../../common/constants';
 import { createPipelineIfNotExists } from './create_processor';
 import { benchmarkScoreMapping } from './benchmark_score_mapping';
 import { latestFindingsPipelineIngestConfig, scorePipelineIngestConfig } from './ingest_pipelines';
+import { latestIndexConfigs } from './latest_indices';
+import { IndexConfig, IndexTemplateParams } from './types';
 
 // TODO: Add integration tests
-
 export const initializeCspIndices = async (esClient: ElasticsearchClient, logger: Logger) => {
   await Promise.all([
     createPipelineIfNotExists(esClient, scorePipelineIngestConfig, logger),
     createPipelineIfNotExists(esClient, latestFindingsPipelineIngestConfig, logger),
   ]);
 
   return Promise.all([
-    createLatestFindingsIndex(esClient, logger),
+    ...createLatestIndices(esClient, logger),
```

Comment:
what are the three dots are using for? 👀

Reply:
The `...` is a[ spread ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)syntax that expands arrays or objects into individual elements. Here, we  are expanding the `createLatestIndices` which is an array of individual promises
Example: 
```
Promise.all([
    createLatestFindingsIndex(esClient, logger),
    ...createLatestIndices(esClient, logger)
])
```
equals 

```
Promise.all([ 
createLatestIndex(esClient, logger, findingsIndexConfig), 
createLatestIndex(esClient, logger, vulnerabilitiesIndexConfig),
createBenchmarkScoreIndex(esClient, logger)
])
```

Reply:
`Promise.allSettled` ++

For this case, I don't recommend spreading `createLatestIndices` as it's increasing the cognitive load to read this recode. I would not mind being verbose if that helps reading:


```js
 const [
    createFindingsLatestIndexPromise,
    createVulnerabilitiesLatestIndexPromise,
    createBenchmarkScoreIndexPromise,
  ] = await Promise.allSettled([
    createLatestIndex(esClient, logger, findingsIndexConfig),
    createLatestIndex(esClient, logger, vulnerabilitiesIndexConfig),
    createLatestIndex(esClient, logger),
  ]);
```

I think this way will be clearer that we are calling 3 Promises.

Also, I think we should prepare to handle each error. We can start with logging the reason in case the promise status is `rejected`.

```js
  if (createFindingsLatestIndexPromise.status === 'rejected') {
    logger.error(createFindingsLatestIndexPromise.reason);
  }
  if (createVulnerabilitiesLatestIndexPromise.status === 'rejected') {
    logger.error(createVulnerabilitiesLatestIndexPromise.reason);
  }
  if (createBenchmarkScoreIndexPromise.status === 'rejected') {
    logger.error(createBenchmarkScoreIndexPromise.reason);
  }
```

That means we can throw the error `increateLatestIndex` and `createLatestIndex` functions instead of logging, and all the logging can be done here, and gradually we can improve the handling for each case.

Reply:
okay I'll change ` ...createLatestIndices(esClient, logger)` since it's adding confusion. It was mainly adds an automated flow if more indices are added to the `latestIndexConfig`.

I believe we are logging an error here ` logger.error(Failed to upsert index template [Template: ${indexTemplateName}]);`  depending on which template fails in `createLatestIndex`. Wouldn't be redundant to log each error again? @opauloh

Reply:
yeah, we can replace the logging with a `throw`

```js
threw new Error(Failed to upsert index template [Template: ${indexTemplateName}]);
```

and then the logging will be done on the error:

```js
if (createFindingsLatestIndexPromise.status === 'rejected') {
  logger.error(createFindingsLatestIndexPromise.reason);
}
```

that should be done for the `createBenchmarkScoreIndexPromise` as well